### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by EnderSmith**
TL;DR:
- REMOVED old commented code in `diceRollUnitTests.html` and `diceRollUnitTests.js`.
- REFACTORED `UnitTest()`:
- - ADDED `try`/`catch`-es in `UnitTest()`.
- - RENAMED existing properties.
- - ADDED new properties.
- - CHANGED `UnitTest(functionToBeTested)` only accepts anonymous functions.
- - ADDED `UnitTest(expectedOutput)` accepts a value or a function.
- ADDED anonymous functions to all instances of `UnitTest(functionToBeTested)` (6).

ADDRESSED AFTER REVIEW by ultimatetag:
- MOVED `UnitTest().functionToBeTested()`&`UnitTest().expectedOutput()` inside `try`/`catch`.
- ADDED brackets to `if` statement in `UnitTest` to avoid mistakes and improve readability.

HTMLpreview:
http://htmlpreview.github.io/?https://github.com/EnderSmith/EnderSmith.github.io/blob/unitTesting/diceRoll/diceRollUnitTests.html

### CODE REVIEW

File: diceRoll/diceRollUnitTests.js [hunk 1]
```diff
@@ -26,15 +25,22 @@ function runTests(runTF) {
...
-    return this.report;
-  } else {
-    this.report = this.functionToBeTested + '\n\tFailed.\n\tReturned: ' + this.functionReturned;
-    console.log(this.report);
-    return this.report;
+  this.testName = testName;
+  this.functionToBeTested = functionToBeTested;
+  this.functionReturned = this.functionToBeTested();
+  this.expectedOutput = expectedOutput;
+  if (typeof this.expectedOutput === 'function') this.expectedOutput = this.expectedOutput();
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.js]: suggestion: even though you can use if statements without { }, its better to use them to avoid mistakes and improve readability

Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.js]: actually, you need to execute the expectedOutput inside the try statement, otherwise there is no reason to use a try/catch.

```
if (typeof this.expectedOutput === 'function') {
  try {
      this.expectedOutput = this.expectedOutput();
      if (this.functionReturned != this.expectedOutput) {
       this.report = '\n\tFailed.\n\tReturned: ' + this.functionReturned + '\n\tExpected: ' + this.expectedOutput;
       throw this.report;
     }
   } catch (error) {
     console.log(this.testName + error);
     return this.testName + this.report;
   }
}
```

**Review by ultimatetag — APPROVED** [review]
I am approving, but you should follow my suggestions before completing
